### ISSUE TITLE
Work with groonga-httpd with BASIC authentication

### ISSUE THREAD

**Issue by yoshihara** [hiroyuki-sato/embulk-output-groonga#1]
（日本語）twitterでご連絡したとおりの内容です。設定内容や手順など書きましたので調査のご参考になれば幸いです。（ファイルは多かったのでgistに入れました）もし何かありましたらこちらのコメントもしくはtwitterにてご連絡いただければ幸いです。

---

Hi,

I use embulk-output-groonga to insert data to groonga-httpd.
The groonga-httpd server has configuration for BASIC authentication, but I can do it without id/pass for BASIC auth unexpectedly. Would you please tell me this is the correct work ?

I prepare files to reproduce this, here https://gist.github.com/yoshihara/bed63fca0f852aed282f93f7c73b9e99 .
This groonga-httpd server has BASIC authentication `groonga` / `test` .

How to reproduce:
```sh
% sudo docker build -f Dockerfile -t yoshihara/groonga-test ./
% sudo docker run -v $PWD:/groonga -p 10041:10041  -d yoshihara/groonga-test
% embulk run embulk_config.yml
```

This is `embulk run` output:

```sh
2017-10-14 19:05:56.101 +0900: Embulk v0.8.34
2017-10-14 19:06:01.892 +0900 [INFO] (0001:transaction): Loaded plugin embulk-output-groonga (0.1.1)
2017-10-14 19:06:01.920 +0900 [INFO] (0001:transaction): Listing local files at directory '.' filtering filename by prefix 'test.csv'
2017-10-14 19:06:01.922 +0900 [INFO] (0001:transaction): "follow_symlinks" is set false. Note that symbolic links to directories are skipped.
2017-10-14 19:06:01.928 +0900 [INFO] (0001:transaction): Loading files [test.csv]
2017-10-14 19:06:01.978 +0900 [INFO] (0001:transaction): Using local thread executor with max_threads=8 / output tasks 4 = input tasks 1 * 4
2017-10-14 19:06:02.023 +0900 [INFO] (0001:transaction): {done:  0 / 1, running: 0}
2017-10-14 19:06:02.721 +0900 [INFO] (embulk-output-executor-0): groonga inserted 3 / 3
2017-10-14 19:06:02.724 +0900 [INFO] (0001:transaction): {done:  1 / 1, running: 0}
2017-10-14 19:06:02.735 +0900 [INFO] (main): Committed.
2017-10-14 19:06:02.736 +0900 [INFO] (main): Next config diff: {"in":{"last_path":"test.csv"},"out":{}}
```

Using Groonga version is 6.1,5. embulk-output-groonga is 0.1.1.
Thanks.

**Comment by hiroyuki-sato** [comment]
Reference: https://twitter.com/hyoshihara04/status/918865916154093570

**Comment by yoshihara** [comment]
This patch makes my expected response by `curl http://localhost:10041/d/status` (gets 401) : 

```diff
diff --git a/groonga-httpd.conf b/groonga-httpd.conf
index 0279737..962f84c 100644
--- a/groonga-httpd.conf
+++ b/groonga-httpd.conf
@@ -44,6 +44,9 @@ http {
     server_name localhost;
 
     location /d/ {
+      auth_basic "Restricted";
+      auth_basic_user_file /etc/nginx/.htpasswd;
+
       groonga on;
       # You can disable log for Groonga.
       groonga_log_path /var/lib/groonga/log/log;
```

After applying this patch, `curl -u groonga:test http://localhost:10041/d/status` gets status info correctly.

**Comment by yoshihara** [comment]
I use embulk-output-groonga v0.1.2 so this problem is fixed. Thanks! :)

**Comment by hiroyuki-sato** [comment]
Thank you!
Fixed by #3
